### PR TITLE
Reject invalid GitHub identifiers at webhook parse

### PR DESCRIPTION
**Pull request by prathamdby**
## Summary

- Webhook parse requires integer PR numbers in the Postgres integer range
- Installation, comment, and user ids must be positive JS-safe integers
- Owner, repo, login, and sha strings must be non-empty and length-capped, including `pull_request.before` and comment user logins
- Intake returns 422 for those parse failures and does not write work rows
- CONTEXT and configuration document the identifier bounds

## Details

- `githubPrNumberSchema`, `githubSafeIdSchema`, `githubUserSchema`, and the string bound pipes live in `src/webhook/payloads/common.ts`
- Pull request, comment, workflow_run, and check_suite schemas use those pipes
- `GITHUB_INT32_ID_MAX` and related caps live in `src/settings/githubConstants.ts`
- `parseGithubPayload` throws `WebhookParseError` for float, zero, negative, overflow, empty, and oversized fields
- `processWebhookPostRequestEffect` returns 422 without scheduler or handler calls for invalid identifiers on pull_request, issue_comment, workflow_run, and check_suite

<div><a href="https://cursor.com/agents/bc-c6706554-a000-4a49-be29-f24941ca4a4a?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-c6706554-a000-4a49-be29-f24941ca4a4a&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### PR Type

Enhancement

### Description

- Add bounded GitHub identifier constants to githubConstants for PR numbers, safe ids, logins, repo names, and shas.
- Replace bare number/string fields in webhook payload schemas with Valibot pipes that enforce integer, range, and length limits.
- Extend validation to pull_request, check_suite, workflow_run, issueComment, and review comment payloads via shared schemas.
- Tighten persistence contract: PR numbers now respect Postgres integer max 2147483647 and safe ids respect MAX_SAFE_INTEGER.
- Map validation failures to 422 unprocessable entity without writing webhook_events or work items, so GitHub does not retry malformed payloads.
- Document new limits and 422 behavior in configuration and agent-work-ops docs, and add parse/pipeline tests for edge cases.

### Review map

1. <a href="https://github.com/prathamdby/pr-agent/pull/470/files#diff-2a233f002c0493f4d98b7a91093a5e6c315e408c81c08cc6e53a1f69980d8118"><code>src/webhook/payloads/common.ts</code></a>: Central bounded schemas that all webhook events depend on
2. <a href="https://github.com/prathamdby/pr-agent/pull/470/files#diff-9453577f596b332b32abfd8264fbf041f0608b4e0cf16506449a812b10615922"><code>src/settings/githubConstants.ts</code></a>: New identifier bounds that define persistence and safety limits
3. <a href="https://github.com/prathamdby/pr-agent/pull/470/files#diff-5f5f9c21004c15471ce6ccdeba27af28802561c0508a33b6c37d64bfdbf9573a"><code>src/webhook/payloads/pullRequestEvent.ts</code></a>: PR number and SHA validation on primary intake path
4. <a href="https://github.com/prathamdby/pr-agent/pull/470/files#diff-a368e4f02d8e5b87fc77674cc9f37ca708f94b8cf76ab294698b66e83eed2fa8"><code>src/webhook/payloads/issueCommentEvent.ts</code></a>: Safe-id checks on comment and user ids from external input
5. <a href="https://github.com/prathamdby/pr-agent/pull/470/files#diff-25b6b93160afb8cf59c0bce996358c672bbfae3e43b8c894e43bf186e0ccc1ea"><code>src/webhook/payloads/checkSuiteEvent.ts</code></a>: Head SHA and id validation for CI-triggered refresh flow

### CODE REVIEW

File: src/webhook/payloads/common.ts [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
+  v.string(),
+  v.minLength(1),
+  v.maxLength(GITHUB_REPO_NAME_MAX_CHARS),
+);
+
+export const githubShaSchema = v.pipe(
+  v.string(),
+  v.minLength(1),
+  v.maxLength(GITHUB_SHA_MAX_CHARS),
+);
```
[the remark this answers is not in the document]
Comment (prathamdby) [reply src/webhook/payloads/common.ts:42]: Rejecting this. `githubShaSchema` is a length fence on intake, not a Git object-ID parser. Checkout already requires a 40-char hex SHA in `assertSha`, so a value like `abc` does not get scheduled as durable review or CI work. Tightening the webhook schema to hex would reject valid short SHAs GitHub still sends. Leaving the length bound here.
